rust/src/DEanalysis.rs

```
@@ -664,6 +664,14 @@ fn main() {
                                 //println!("filtering time:{:?}", filtering_time.elapsed());
                                 //println!("filtered_matrix_rows:{:?}", filtered_matrix.nrows());
                                 //println!("filtered_matrix_cols:{:?}", filtered_matrix.ncols());
+                                if filtered_matrix.nrows() == 0 {
+                                    // Its possible after filtering there might not be any genes left in the matrix, in such a case the rust code must exit gracefully with an error.
+                                    panic!("Number of genes after filtering = 0, cannot proceed any further")
```

I like this. My only concern is I don't think panic would return output to a user so it may fail silently. We might want to instead encapsulate the messages about no genes or samples after filtering into json objects that get returned and get displayed to the user in the console

---

@robinpaul85 Sorry this slipped by me. That sounds good